### DISCUSSION TITLE
Having trouble connecting to database in a docker install

### DISCUSSION THREAD

**Discussion by sergio101** [mastodon/mastodon#29901] [Q&A]
I am having issues with database connection when running mastodon in a docker container.

During the setup process, I set up a mastodon user and gave them permission to create a database.

During the setup process, the database setup works, and the process finishes.

When I do a `docker compose up`, I get the following error:

    db-1         | 2024-04-10 21:06:52.764 UTC [29] FATAL:  password authentication failed for user "mastodon"
    db-1         | 2024-04-10 21:06:52.764 UTC [29] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"

and

    web-1        | /opt/mastodon/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.8.1/lib/active_record/connection_adapters/postgresql_adapter.rb:81:in `rescue in new_client': We could not find your database: mastodon. Which can be found in the database configuration file located at config/database.yml. (ActiveRecord::NoDatabaseError)
    web-1        |
    web-1        | To resolve this issue:
    web-1        |
    web-1        | - Did you create the database for this app, or delete it? You may need to create your database.
    web-1        | - Has the database name changed? Check your database.yml config has the correct database name.
    web-1        |
    web-1        | To create your database, run:
    web-1        |
    web-1        |         bin/rails db:create

Connecting to DB container shows that the database and user do exist.

The password I used when creating the user is the same as the one in `.env.production`

The db part of my docker-container looks like

    services:
      db:
        restart: always
        image: postgres:14-alpine
        shm_size: 256mb
        networks:
          - internal_network
        healthcheck:
          test: ['CMD', 'pg_isready', '-U', 'postgres']
        volumes:
          - ./postgres14:/var/lib/postgresql/data
        environment:
          - 'POSTGRES_HOST_AUTH_METHOD=trust'

Any ideas?

Thanks!

**Comment by mjankowski** [comment]
Did you ever sort this out?

If not, did you double check that all the DB username, password, and database name do match? (ie `mastodon_production` vs `mastodon` or something...)
